docs: Add Google/AWS/Azure test blog (#1282)
iresty/iresty.github.io: blog/en/blog/2022/08/12/arm-performance-google-aws-azure-with-apisix.md

---

> title: "ARM is trending! Performance between Google&AWS&Azure"

Reviewer: ```suggestion
title: "ARM is trending! Performance Comparisons among Google, AWS, and Azure"
```

authors:
  - name: "Shirui Zhao"
    title: "Author"
    url: "https://github.com/soulbird"
    image_url: "https://github.com/soulbird.png"
  - name: "Sylvia"
    title: "Technical Writer"
    url: "https://github.com/SylviaBABY"
    image_url: "https://avatars.githubusercontent.com/u/39793568?v=4"
keywords: 
- API gateway
- ARM
- Azure
- AWS
- Google
- Apache APISIX

> description: This article uses API Gateway Apache APISIX to compare the performance in ARM between Google&AWS&Azure in network IO-intensive scenarios.

Reviewer: ```suggestion
description: This article uses API Gateway Apache APISIX to compare the performance in ARM among Google, AWS, and Azure in network IO-intensive scenarios.
```

tags: [Ecosystem]
---

> > This article uses API Gateway Apache APISIX to compare the performance in ARM between Google&AWS&Azure in network IO-intensive scenarios.

Reviewer: ditto

<!--truncate-->

<head>
    <link rel="canonical" href="https://api7.ai/2022/08/12/arm-performance-google-aws-azure-with-apisix/" />
</head>

## Background

The ARM architecture belongs to the [RISC](https://en.wikipedia.org/wiki/Reduced_instruction_set_computer). The RISC microprocessor architecture design uses a highly optimized set of instructions to enable small processors to efficiently handle complex tasks.

ARM has become the cornerstone of the world's largest computing ecosystem and mobile devices, and is considered by many experts to be the future of cloud computing due to its low power consumption, flexible licensing and low cost.

Therefore, mainstream cloud vendors led by AWS, Google Cloud Platform (GCP) and Azure have successively launched servers with ARM architecture. Among them, AWS launched the first server processor AWS Graviton based on ARM architecture in 2018.

### AWS Graviton

> AWS Graviton is a series of server processors based on the [ARM architecture](https://www.arm.com/) released by AWS in 2018\. The first generation of AWS Graviton processors uses custom chips and 64-bit Neoverse cores.

Reviewer 2: ```suggestion
AWS Graviton is a series of server processors based on the [ARM architecture](https://www.arm.com/) released by AWS in 2018. The first generation of AWS Graviton processors uses custom chips and 64-bit Neoverse cores.
```

Released in 2020, AWS Graviton2 processors represent a major leap forward in performance and functionality compared to first-generation AWS Graviton processors. 7x faster performance, 4x more cores, 2x cache, 5x faster memory, and more.

> The latest AWS Graviton3 processors to be released at the end of May 2022 are based on the more advanced Neoverse V1 design, they offer up to twice the floating point performance, twice the cryptographic performance, and three times the ML compared to the AWS Graviton2 processors performance, including support for bfloat16\. The following figure shows the main models equipped with AWS Graviton3 processors:

Reviewer 2: ```suggestion
The latest AWS Graviton3 processors to be released at the end of May 2022 are based on the more advanced [Neoverse V1](https://www.arm.com/zh-TW/products/silicon-ip-cpu/neoverse/neoverse-v1) design, they offer up to twice the floating point performance, twice the cryptographic performance, and three times the ML compared to the AWS Graviton2 processors performance, including support for bfloat16. The following figure shows the main models equipped with AWS Graviton3 processors:
```

| VM series / vCPU | 1       | 2       | 4       | 8      | 16      | 32      | 64      |
|------------------|---------|---------|:--------|:-------|:--------|:--------|:--------|
| AWS C7g          | $0.0361 | $0.0723 | $0.1445 | $0.289 | $0.5781 | $1.1562 | $1.7342 |
| GCP T2A          | $0.0385 | $0.077  | $0.154  | $0.308 | $0.616  | $1.232  | $1.848  |
| Azure Dpsv5      | *       | $0.077  | $0.154  | $0.308 | $0.616  | $1.232  | $1.848  |

The following table summarizes the cost and cost–performance ratio of AWS c7g.large and GCP t2a-standard-2 running for one year, referring to the QPS data for a single upstream in the Apache APISIX performance test. The larger the number, the higher the QPS can be obtained at the unit price.

|                    | Annual cost         | Cost performance (QPS/cost) |
|--------------------|-----------------|--------------------|
| AWS c7g.large      | $633.3          | 36.3               |
| GCP t2a-standard-2 | $674.5          | 16.8               |
| Azure D2ps v5      | $398.0（41% off) | 33.6               |

> From the test results, AWS C7g is more cost-effective than GCP T2A and Azure Dpsv5\. Although Azure Dpsv5 has only a 15% performance improvement compared to GCP T2A, the cost performance is nearly double.

Reviewer 2: ```suggestion
From the test results, AWS C7g is more cost-effective than GCP T2A and Azure Dpsv5. Although Azure Dpsv5 has only a 15% performance improvement compared to GCP T2A, the cost performance is nearly double.
```

## Summary

> AWS launched the first ARM-based processor, AWS Graviton, in 2018\. It was about 4 years ahead of GCP for the deployment of the ARM-based server field. Now the AWS Graviton processor has developed to the third generation.

Reviewer 2: ```suggestion
AWS launched the first ARM-based processor, AWS Graviton, in 2018. It was about 4 years ahead of GCP for the deployment of the ARM-based server field. Now the AWS Graviton processor has developed to the third generation.
```

> Through the performance test results and price–performance ratio analysis of Apache APISIX, we can see that AWS Graviton3 has higher performance and cost-effective than GCP T2A and Azure Dpsv5\. This is inseparable from the fact that AWS has been deeply involved in the field of ARM-based servers for many years.

Reviewer 2: ```suggestion
Through the performance test results and price–performance ratio analysis of Apache APISIX, we can see that AWS Graviton3 has higher performance and cost-effective than GCP T2A and Azure Dpsv5. This is inseparable from the fact that AWS has been deeply involved in the field of ARM-based servers for many years.
```
